amps/ags/rm-community/rm-community-repo/source/java/org/alfresco/rm/rest/api/impl/ApiNodesModelFactory.java

```diff
@@ -1033,4 +1069,81 @@ public void mapRetentionScheduleOptionalInfo(RetentionSchedule retentionSchedule
             retentionSchedule.setActions(actions);
         }
     }
+
+    /**
+     * this method is used for creation of retention schedule action definition params
+     * @param nodeInfo retention schedule action definition
+     * @return Map<QName, Serializable>
+     */
+    public Map<QName, Serializable> createRetentionActionDefinitionParams(RetentionScheduleActionDefinition nodeInfo)
+    {
+        Map<QName, Serializable>  actionDefinitionParams= new HashMap<>();
+        actionDefinitionParams.put(RecordsManagementModel.PROP_DISPOSITION_ACTION_NAME, nodeInfo.getName());
+        actionDefinitionParams.put(RecordsManagementModel.PROP_DISPOSITION_DESCRIPTION, nodeInfo.getDescription());
+        StringBuilder retentionPeriod = new StringBuilder(nodeInfo.getPeriod()).append("|");
+        if(isPeriodAmountApplicable(nodeInfo.getPeriod()))
+        {
+            retentionPeriod.append(nodeInfo.getPeriodAmount());
+        }
+        actionDefinitionParams.put(RecordsManagementModel.PROP_DISPOSITION_PERIOD, retentionPeriod.toString());
+        QName periodProperty = QName.createQName(nodeInfo.getPeriodProperty(), namespaceService);
+        actionDefinitionParams.put(RecordsManagementModel.PROP_DISPOSITION_PERIOD_PROPERTY, periodProperty);
+        actionDefinitionParams.put(RecordsManagementModel.PROP_DISPOSITION_EVENT_COMBINATION,
+                nodeInfo.isEligibleOnFirstCompleteEvent());
+        actionDefinitionParams.put(RecordsManagementModel.PROP_COMBINE_DISPOSITION_STEP_CONDITIONS,
+                nodeInfo.isCombineDispositionStepConditions());
+        actionDefinitionParams.put(RecordsManagementModel.PROP_DISPOSITION_LOCATION,
+                nodeInfo.getLocation());
+        List<String> inputEvents = nodeInfo.getEvents();
+        actionDefinitionParams.put(RecordsManagementModel.PROP_DISPOSITION_EVENT, (Serializable) inputEvents);
+        if (RetentionSteps.DESTROY.stepName.equals(nodeInfo.getName()) && nodeInfo.isRetainRecordMetadataAfterDestruction())
+        {
+            actionDefinitionParams.put(RecordsManagementModel.PROP_DISPOSITION_ACTION_GHOST_ON_DESTROY, "ghost");
+        }
+        else
+        {
+            actionDefinitionParams.put(RecordsManagementModel.PROP_DISPOSITION_ACTION_GHOST_ON_DESTROY, "delete");
+        }
+        return actionDefinitionParams;
+    }
+
+    /**
+     * this method is used retrieve retention schedule action details
+     * @param retentionScheduleNodeRef nodeRef
+     * @return List<DispositionActionDefinition>
+     */
+    public List<DispositionActionDefinition> getRetentionActions(NodeRef retentionScheduleNodeRef)
+    {
+        List<ChildAssociationRef> assocs = nodeService.getChildAssocs(
+                retentionScheduleNodeRef,
+                RecordsManagementModel.ASSOC_DISPOSITION_ACTION_DEFINITIONS,
+                RegexQNamePattern.MATCH_ALL);
+        List<DispositionActionDefinition> actions;
+        actions = IntStream.range(0, assocs.size())
```

Comment:
Put the comment as what is this code doing.

Reply:
done.